tests/ui/drop/if-let-rescope-suggestions.rs

```diff
@@ -0,0 +1,25 @@
+//@ edition:2024
```

Comment:
Problem [REV7]: this needs to check < Edition 2024 as we need to test the migration lint.

Reply:
My bad. This test should be renamed `if-let-rescope-borrowck-suggestions` because it is testing this scenario. When Edition 2024 is enabled and `if_let_rescope` rule is enabled as well, if a borrow checking error is found and originated from `if let`, it will emit a suggestion to rewrite `if let` with `match`. It is different from Edition migration.

Reply:
Ah ok, thanks for the clarification.